app/src/main/java/com/coopra/nebulus/view_models/AlternativeViewModel.java

```diff
@@ -0,0 +1,51 @@
+package com.coopra.nebulus.view_models;
+
+import android.app.Application;
+import android.os.AsyncTask;
+
+import com.coopra.data.Track;
+import com.coopra.nebulus.TrackRepository;
+
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.MutableLiveData;
+
+public class AlternativeViewModel extends AndroidViewModel {
+    private MutableLiveData<List<Track>> mTracks;
+
+    public AlternativeViewModel(@NonNull Application application) {
+        super(application);
+    }
+
+    public MutableLiveData<List<Track>> getTracks() {
+        if (mTracks == null) {
+            mTracks = new MutableLiveData<>();
+        }
+
+        return mTracks;
+    }
+
+    public void retrieveTracks() {
+        new GetTracksTask(getTracks()).execute();
```

Comment:
Now that `mTracks` will never be null, you can use it directly here. You may also want to consider only launching a new task here if you think you can actually get more data - not sure if you have a way to know that or not...

```suggestion
        new GetTracksTask(mTracks).execute();
```